acvm-repo/acvm/tests/solver.rs

```diff
@@ -841,9 +1104,161 @@ prop_compose! {
+    assert_eq!(result, expected_result);
+}
+
+#[test]
+fn sha256_zeros() {
+    let results = solve_array_input_blackbox_call(vec![], 32, sha256_op);
+    let expected_results: Vec<_> = vec![
+        227, 176, 196, 66, 152, 252, 28, 20, 154, 251, 244, 200, 153, 111, 185, 36, 39, 174, 65,
+        228, 100, 155, 147, 76, 164, 149, 153, 27, 120, 82, 184, 85,
+    ]
+    .into_iter()
+    .map(|x: u128| FieldElement::from(x))
+    .collect();
+    assert_eq!(results, expected_results);
+}
+
+#[test]
+fn sha256_compression_zeros() {
+    let results = solve_array_input_blackbox_call(
+        [(FieldElement::zero(), false); 24].try_into().unwrap(),
+        8,
+        sha256_compression_op,
+    );
+    let expected_results: Vec<_> = vec![
+        2091193876, 1113340840, 3461668143, 3254913767, 3068490961, 2551409935, 2927503052,
+        3205228454,
+    ]
+    .into_iter()
+    .map(|x: u128| FieldElement::from(x))
+    .collect();
+    assert_eq!(results, expected_results);
+}
+
+#[test]
+fn blake2s_zeros() {
+    let results = solve_array_input_blackbox_call(vec![], 32, blake2s_op);
+    let expected_results: Vec<_> = vec![
+        105, 33, 122, 48, 121, 144, 128, 148, 225, 17, 33, 208, 66, 53, 74, 124, 31, 85, 182, 72,
+        44, 161, 165, 30, 27, 37, 13, 253, 30, 208, 238, 249,
+    ]
+    .into_iter()
+    .map(|x: u128| FieldElement::from(x))
+    .collect();
+    assert_eq!(results, expected_results);
+}
+
+#[test]
+fn blake3_zeros() {
+    let results = solve_array_input_blackbox_call(vec![], 32, blake3_op);
+    let expected_results: Vec<_> = vec![
+        175, 19, 73, 185, 245, 249, 161, 166, 160, 64, 77, 234, 54, 220, 201, 73, 155, 203, 37,
+        201, 173, 193, 18, 183, 204, 154, 147, 202, 228, 31, 50, 98,
+    ]
+    .into_iter()
+    .map(|x: u128| FieldElement::from(x))
+    .collect();
+    assert_eq!(results, expected_results);
+}
+
+#[test]
+fn keccak256_zeros() {
+    let results = solve_array_input_blackbox_call(vec![], 32, keccak256_op);
+    let expected_results: Vec<_> = vec![
+        197, 210, 70, 1, 134, 247, 35, 60, 146, 126, 125, 178, 220, 199, 3, 192, 229, 0, 182, 83,
+        202, 130, 39, 59, 123, 250, 216, 4, 93, 133, 164, 112,
+    ]
+    .into_iter()
+    .map(|x: u128| FieldElement::from(x))
+    .collect();
+    assert_eq!(results, expected_results);
+}
+
+// TODO(https://github.com/noir-lang/noir/issues/5687): internal error when calling Keccakf1600
+#[test]
+#[should_panic(expected = "assertion `left == right` failed")]
+fn keccakf1600_zeros() {
+    assert_eq!(FieldElement::zero().num_bits(), 0);
+
+    let _results = solve_array_input_blackbox_call(
+        [(FieldElement::zero(), false); 25].into(),
+        25,
+        keccakf1600_op,
+    );
+    // TODO(https://github.com/noir-lang/noir/issues/5687): re-enable once internal error is resolved
+    // assert_eq!(results, vec![]);
+}
+
+// TODO(https://github.com/noir-lang/noir/issues/5690): expected to be injective
```

Comment:
I think this is also due to FunctionInputs always using `max_num_bits` rather than the appropriate bit size.

Reply:
We should have a clearer ban on the bit size for these functions so this isn't possible at all. Let's make sure we are setting up the input bits correctly here and update the issue accordingly. At a minimum we should panic if the incorrect bit size is provided (e.g. 254 instead of 8 for keccak256) so that we know codegen is incorrect.

Reply:
We could probably just return an `OpcodeResolutionError<F>::BlackBoxFunctionFailed`

Reply:
Ok, this is now passing. Made a follow-up issue for the feature: https://github.com/noir-lang/noir/issues/5793

Reply:
This issue https://github.com/noir-lang/noir/issues/5690 can be closed then?